Source/StyledTextView.swift

```diff
@@ -171,10 +174,20 @@ open class StyledTextView: UIView {
     }
 
     open func reposition(for width: CGFloat) {
-        guard let renderer = self.renderer else { return }
-        let result = renderer.render(for: width)
-        layer.contents = result.image
-        frame = CGRect(origin: CGPoint(x: renderer.inset.left, y: renderer.inset.top), size: result.size)
+        guard let capturedRenderer = self.renderer else { return }
+        if drawsAsync {
+            StyledTextView.renderQueue.async {
+                let result = capturedRenderer.render(for: width)
+                DispatchQueue.main.async {
+                    if capturedRenderer !== self.renderer { return }
```

Comment:
This provides the safety from races. If the renderer changes, then we abort. However, this depends on the renderers being immutable. Is that the case @rnystrom?

Reply:
Pretty much. The only mutable things are the `NSLayoutManager` and `NSTextContainer` which are `internal` to the framework.

https://github.com/GitHawkApp/StyledTextKit/blob/master/Source/StyledTextRenderer.swift#L13-L14

I guess the only place this might cause trouble is w/ resizing. The renderer doesn't change but the size of the container is mutated.

https://github.com/GitHawkApp/StyledTextKit/blob/master/Source/StyledTextRenderer.swift#L74

Maybe we should check that the `bounds.width` hasn't changed too?